### PR TITLE
Remove synchronized from QueueFile

### LINKED ISSUE #106: Remove synchronized from QueueFile methods?
Like most collections, we could just document that the class isn't thread safe and rely on consumers to synchronize on the instance if they're doing multithreading.

### PR DESCRIPTION
**Pull request by Macarse**
Fixes #106

### CODE REVIEW

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 1]
```diff
@@ -30,13 +30,15 @@
 
 /**
  * A reliable, efficient, file-based, FIFO queue. Additions and removals are O(1). All operations
- * are atomic. Writes are synchronous; data will be written to disk before an operation returns.
+ * are atomic. Data will be written to disk before an operation returns.
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: > Writes are synchronous...

This still holds, I don't think the docs need to be updated.

Comment (JakeWharton) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: Agreed. Writes remain synchronous.

Comment (Macarse) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: fixed.

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 2]
```diff
@@ -35,8 +35,10 @@
  * exception is thrown during a mutating change, the change is aborted. It is safe to continue to
  * use a {@code QueueFile} instance after an exception.
  *
- * <p>All operations are synchronized. In a traditional queue, the remove operation returns an
- * element. In this queue, {@link #peek} and {@link #remove} are used in conjunction. Use
+ * <p><strong>Note that this implementation is not synchronized.</strong>
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: I wonder if this note needs to go at the class level, not just for `add`?

Comment (Macarse) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: @f2prateek it's at the class level. Are you confused by the diff showing a jump from line 44 to 300 where `add()` javadoc starts?

Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: Ah yup that confused me!

**Comment by f2prateek** [comment]
LGTM, thanks!

**Pull request merged**